ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -110,9 +115,23 @@ export default class ConfirmPageContainer extends Component {
     supportsEIP1559V2: PropTypes.bool,
     nativeCurrency: PropTypes.string,
     isBuyableChain: PropTypes.bool,
-    isApprovalOrRejection: PropTypes.bool,
   };
 
+  async componentDidMount() {
```

Comment:
@pedronfigueiredo, @bschorchit @amerkadicE -- Love this suggestion considering this component is low complexity. I'm in favor of either

A) blocking PR until this is completed as we are introducing a componentDidMount here which actually **adds** complexity for a future PR to convert this to functional.
B) Freeing @amerkadicE up to do this work in a PR immediately to follow this one. 

Either option helps to pay off some debt instead of incurring new debt.

Reply:
> @pedronfigueiredo, @bschorchit @amerkadicE -- Love this suggestion considering this component is low complexity. I'm in favor of either
> 
> 
> 
> A) blocking PR until this is completed as we are introducing a componentDidMount here which actually **adds** complexity for a future PR to convert this to functional.
> 
> B) Freeing @amerkadicE up to do this work in a PR immediately to follow this one. 
> 
> 
> 
> Either option helps to pay off some debt instead of incurring new debt. 

I vote B so the user facing changes (added friction for this request) reach the users sooner :) 
I'm happy to create the ticket for the refactoring so it gets worked on next.

Reply:
I agree I think option B works!

Reply:
For visibility, new ticket created here: https://github.com/MetaMask/metamask-extension/issues/16824

Reply:
You can assign it to me @bschorchit

Reply:
great, thanks all